hub/api/dataset.py

```diff
@@ -5,29 +5,62 @@
     InvalidKeyTypeError,
     UnsupportedTensorTypeError,
 )
-from typing import Union, Dict
+from hub.core.storage.provider import StorageProvider
```

Comment:
out of scope, but @AbhinavTuli we should change these imports for storage to be:

```
from hub.core.storage import StorageProvider, MemoryProvider, LocalProvider
```

Reply:
oh wait btw this is already implemented in `pytest-options` looool